### PR TITLE
DOCSP-39542: delete crud fundamentals

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-rust/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-39542>
Staging - 
https://docs-mongodbcom-staging.corp.mongodb.com/rust/docsworker-xlarge/DOCSP-39542-delete/fundamentals/crud/write-operations/delete/
https://docs-mongodbcom-staging.corp.mongodb.com/rust/docsworker-xlarge/DOCSP-39542-delete/fundamentals/crud/write-operations/#std-label-rust-crud-write-operations

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/crud/write-operations/delete.txt [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+.. list-table::
+   :header-rows: 1
+   :class: compatibility-large
+
+   * - Option
+     - Description
+
+   * - ``collation`` 
+     - | The collation to use when sorting results.
```
Comment (nickldp) [inline source/fundamentals/crud/write-operations/delete.txt:66]: S: A link to the [Collation](https://www.mongodb.com/docs/manual/reference/collation/) manual page could be helpful here

File: source/fundamentals/crud/write-operations/delete.txt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about any of the methods or types discussed in this
+guide, see the following API Documentation:
+
+- `delete_one() <{+api+}/struct.Collection.html#method.delete_one>`__
+- `delete_many() <{+api+}/struct.Collection.html#method.delete_many>`__
+- `DeleteOptions <{+api+}/options/struct.DeleteOptions.html>`__
+- `Hint <{+api+}/options/enum.Hint.html>`__
```
Comment (nickldp) [inline source/fundamentals/crud/write-operations/delete.txt:174]: S: A link to the API documentation for the let_vars option could be useful here

**Review by nickldp — APPROVED** [review]
LGTM

File: source/fundamentals/crud/write-operations/delete.txt (hunk 1 above)
Comment (rustagir) [reply source/fundamentals/crud/write-operations/delete.txt:66]: Once the collation page for Rust is written, I'll link it to resolve the TODO underneath the table!

File: source/includes/fundamentals/code-snippets/crud/delete.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    let hint = Hint::Name("_id_".to_string());
+    let opts: DeleteOptions = DeleteOptions::builder().hint(hint).build();
+
+    let res = my_coll.delete_many(filter, opts).await?;
+    println!("Deleted documents: {}", res.deleted_count);
+    // end-delete
+
+    // begin-options
+    let opts: DeleteOptions = DeleteOptions::builder().comment(bson!("hello!")).build();
+    let _res = my_coll.delete_one(filter, opts).await?;
```
Comment (kevinAlbs) [inline source/includes/fundamentals/code-snippets/crud/delete.rs:23]: Suggest removing `_` prefix.

**Review by kevinAlbs — COMMENTED** [review]
LGTM with minor suggested change.

**Pull request merged**